fre/yamltools/combine_yamls.py

```diff
@@ -74,193 +83,278 @@ def experiment_check(mainyaml_dir,comb,experiment):
+    Combine the model, compile, and platform yamls
     Arguments:
-        file  :  combined yaml file
+        - comb : combined yaml object 
     """
-    # Load the json schema: .load() (vs .loads()) reads and parses the json in one
-    with open(schema_path) as s:
-        schema = json.load(s)
+    # Merge model into combined file
+    comb_model = comb.combine_model()
+    # Merge compile.yaml into combined file
+    comb_compile = comb.combine_compile()
+    # Merge platforms.yaml into combined file
+    full_combined = comb.combine_platforms()
+    # Clean the yaml
+    full_combined = comb.clean_yaml()
 
-    # Validate yaml
-    # If the yaml is not valid, the schema validation will raise errors and exit
-    if validate(instance=file,schema=schema) is None:
-        print("YAML VALID")
+    return full_combined
 
-###### MAIN #####
-def _consolidate_yamls(yamlfile,experiment, platform,target):
+def get_combined_ppyaml(comb):
     """
-    Process to combine and validate the yamls
+    Combine the model, experiment, and analysis yamls
+    Arguments:
+        - comb : combined yaml object
     """
-    # Regsiter tag handler
-    yaml.add_constructor('!join', join_constructor)
-
-    # Path to the main model yaml
-    mainyaml_dir = os.path.dirname(yamlfile)
-
-    # Name of the combined yaml
-    combined=f"combined-{experiment}.yaml"
-
-    print("Combining yaml files: ")
-
     # Merge model into combined file
-    combine_model(yamlfile,combined,experiment,platform,target)
-
-    # Experiment check
-    (py_path,cy_path,ey_path,ay_path) = experiment_check(mainyaml_dir,combined,experiment)
-
-    # Merge compile.yaml into combined file
-    combine_compile(combined,cy_path)
-
-    # Merge platforms.yaml into combined file
-    combine_platforms(combined,py_path)
-
+    comb_model = comb.combine_model()
     # Merge pp experiment yamls into combined file
-    combine_experiments(combined,ey_path)
-
+    comb_exp = comb.combine_experiment()
     # Merge pp analysis yamls, if defined, into combined file
-    combine_analysis(combined,ay_path)
+    comb_analysis = comb.combine_analysis()
+    # Clean the yaml
+    full_combined = comb.clean_yaml()
 
-    # Load the fully combined yaml
-    full_yaml = yaml_load(combined)
+    return full_combined
 
-    # Clean the yaml
-    # If keys exists, delete:
-    keys_clean=["fre_properties", "shared", "experiments"]
-    for kc in keys_clean:
-        if kc in full_yaml.keys():
-            del full_yaml[kc]
+###########################################################################################
+def _consolidate_yamls(yamlfile,experiment,platform,target,use):
+    # Regsiter tag handler
+    yaml.add_constructor('!join', join_constructor)
+
+    # Path to the main model yaml
+    mainyaml_dir = os.path.dirname(yamlfile)
 
-    with open(combined,'w',encoding='UTF-8') as f:
-        yaml.safe_dump(full_yaml,f,sort_keys=False)
+    if use == "compile":
```

Comment:
im not convinced this is the best variable name here.  maybe we can bring it up at a fre meeting.